### PR TITLE
Fix location provider

### LINKED ISSUE #911: Transportr 2.2.0 crashes after initial setup.
**Describe the bug**
After initial start of the app and selecting any transport network from Germany the app crashes right after granting location permission. It then crashes immediately on every subsequent launching of Transportr. Transportr has been has been fresh installed to verify it's not an update  issue and old data.

**To Reproduce**
Launch the app, select a network and grant location permission. The app will now crash subsequently. 

**Expected behavior**
Normal usage of the app as in the previous version 2.1.5.

**Screenshots**
No screenshots. 

**Versions (please complete the following information):**
 - **Transportr Version:** 2.2.0
 - **Device:** OnePlus 8T
 - **Android Version:** 13.

Crash log:

```
type: crash
osVersion: OnePlus/OnePlus8T/OnePlus8T:13/TQ3A.230901.001/eng.emy.20240107.100827:user/release-keys
package: de.grobox.liberario:124
process: de.grobox.liberario

java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=65537, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {de.grobox.liberario/de.grobox.transportr.map.MapActivity}: java.lang.IllegalArgumentException: provider "network" does not exist
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5339)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5378)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7924)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
Caused by: java.lang.IllegalArgumentException: provider "network" does not exist
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3015)
	at android.os.Parcel.createException(Parcel.java:2995)
	at android.os.Parcel.readException(Parcel.java:2978)
	at android.os.Parcel.readException(Parcel.java:2920)
	at android.location.ILocationManager$Stub$Proxy.registerLocationListener(ILocationManager.java:1306)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1564)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1247)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1212)
	at de.grobox.transportr.map.PositionController.onActive(PositionController.kt:84)
	at de.grobox.transportr.utils.NotifyingLiveData.onActive(NotifyingLiveData.kt:45)
	at androidx.lifecycle.LiveData.changeActiveCounter(LiveData.java:390)
	at androidx.lifecycle.LiveData$ObserverWrapper.activeStateChanged(LiveData.java:466)
	at androidx.lifecycle.LiveData$LifecycleBoundObserver.onStateChanged(LiveData.java:425)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:354)
	at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:196)
	at androidx.lifecycle.LiveData.observe(LiveData.java:205)
	at de.grobox.transportr.map.GpsMapFragment.activateLocationComponent$lambda-6(GpsMapFragment.kt:139)
	at de.grobox.transportr.map.GpsMapFragment.$r8$lambda$bph-Ts8SjpH0ljE8jOsQdWUO9Jw(GpsMapFragment.kt:0)
	at de.grobox.transportr.map.GpsMapFragment$$ExternalSyntheticLambda0.onStyleLoaded(R8$$SyntheticClass:0)
	at com.mapbox.mapboxsdk.maps.MapboxMap.getStyle(MapboxMap.java:120)
	at de.grobox.transportr.map.GpsMapFragment.activateLocationComponent(GpsMapFragment.kt:113)
	at de.grobox.transportr.map.GpsMapFragment.tryActivateLocationComponent(GpsMapFragment.kt:104)
	at de.grobox.transportr.map.GpsMapFragment.onRequestPermissionsResult(GpsMapFragment.kt:177)
	at androidx.fragment.app.FragmentActivity.onRequestPermissionsResult(FragmentActivity.java:769)
	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8811)
	at android.app.Activity.dispatchActivityResult(Activity.java:8669)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5332)
	... 13 more
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:59)
	at com.android.server.location.LocationManagerService.registerLocationListener(LocationManagerService.java:727)
	at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:590)
	at android.os.Binder.execTransactInternal(Binder.java:1280)
	at android.os.Binder.execTransact(Binder.java:1244)


```

### PR DESCRIPTION
**Pull request by Altonss**
This is a quick fix that should close #911.
@Kelvino9 @licaon-kter could you try it out?

### CODE REVIEW

**Comment by licaon-kter** [comment]
rebuilt and confirm that it fixes the issue, no crash

**Comment by Kelvino9** [comment]
Thank you @Altonss for the fix and @licaon-kter for providing an APK.

Transportr does not crash upon start with this build, but also no longer is able to get a fix on my positon like in previous releases. Is there no way to get that function back? 

I do not depend on this functionality as I can use any navigation app to get more information on my surroundings on an unknown area, but it would be a nice to have feature since it was successfully included before.

Thanks anyway for the immediate bug fix.

**Comment by licaon-kter** [comment]
@Kelvino9 so you say that if you lock GPS location in GPSTest app, and then you open this app, it won't get your location?

Works fine for me here.

**Comment by Kelvino9** [comment]
@licaon-kter I deleted everything, rebooted and tested again. It seems to work as intended after waiting for a few moments.

I did install GPSTest this time, because you mentioned it above. It worked. I then uninstalled GPSTest and reinitialized Transportr and it again worked as intended. I hope there isn't any data left from GPSTest and Transportr will continue to work without the need of running GPSTest before.

Was I just impatient when trying to get a fix on my position in Trasnportr? I didn't got a warning from Transportr earlier that it couldn't retrieve my position yet, but it notified me in later tests which actually led me to wait a bit.

**Comment by licaon-kter** [comment]
GPS takes time to get a lock, I guess this app needs a bigger icon for _"looking for location"_ :)

File: app/src/main/java/de/grobox/transportr/map/PositionController.kt [hunk 1]
```diff
@@ -78,10 +78,16 @@ import java.util.concurrent.TimeUnit
             PositionState.DISABLED
     }
 
+     private fun isProviderEnabled(provider: String): Boolean {
+         return locationManager.isProviderEnabled(provider)
+     }
+
     @RequiresPermission(ACCESS_FINE_LOCATION)
     override fun onActive() {
         for (provider in LOCATION_PROVIDERS) {
```
Comment (ialokim) [inline app/src/main/java/de/grobox/transportr/map/PositionController.kt:87]:
```suggestion
        for (provider in LOCATION_PROVIDERS.filter(locationManager::isProviderEnabled)) {
```
and remove the extra if-statement and indentation. Slightly more idiomatic I'd say.

**Review by ialokim — CHANGES_REQUESTED** [review]
Thanks @Altonss for the quick investigation and fix! I've looked into the `PositionController` again and also found the for-loop in `onInactive()` to be useless. Maybe you can remove that too in this PR.

File: app/src/main/java/de/grobox/transportr/map/PositionController.kt (hunk 1 above)
Comment (ialokim) [reply app/src/main/java/de/grobox/transportr/map/PositionController.kt:87]: And the extra private fun, obviously.

**Comment by Kelvino9** [comment]
In Organics Maps you'll be notified to change location permission manually once you denied location permission.

**Comment by ialokim** [comment]
> Only little issue I have, but idk if it's supposed to be normal, is that Transportr never asks me again about the location permission once I refuse (even if I click on the grey locate button).

Good remark, let's follow up on this in #916 .

**Review by ialokim — APPROVED** [review]
Thanks!

**Pull request merged**